NavigationReactNative/sample/twitter/App.js

```diff
@@ -24,29 +24,40 @@ timeline.renderScene = ({id}) => <Timeline timeline={getTimeline(id)}  />;
 var notificationsNavigator = new StateNavigator(stateNavigator);
 stateNavigator.navigate('home');
 
-export default () => (
-  Platform.OS === 'ios' ? (
-    <TabBarIOS>
-      <TabBarItemIOS title="Home">
-        <NavigationHandler stateNavigator={stateNavigator}>
-          <NavigationStack />
-        </NavigationHandler>
-      </TabBarItemIOS>
-      <TabBarItemIOS title="Notifications" onPress={() => {
-        if (!notificationsNavigator.stateContext.state)
-          notificationsNavigator.navigate('notifications');
-      }}>
-        <NavigationHandler stateNavigator={notificationsNavigator}>
-          <NavigationStack />
-        </NavigationHandler>
-      </TabBarItemIOS>
-    </TabBarIOS>
-  ) : (
-    <NavigationHandler stateNavigator={stateNavigator}>
-      <NavigationStack
-        crumbStyle={from => from ? 'scale_in' : 'scale_out'}
-        unmountStyle={from => from ? 'slide_in' : 'slide_out'}>
-      </NavigationStack>
-    </NavigationHandler>
-  )
-);
+export default() => {
+  const [isNotified,
+    setIsNotified] = useState(false);
+  return (Platform.OS === 'ios'
+    ? (<TabBarIOS>
+        <TabBarItemIOS title="Home">
+          <NavigationHandler stateNavigator={stateNavigator}>
+            <NavigationStack/>
+          </NavigationHandler>
+        </TabBarItemIOS>
+        <TabBarItemIOS
+          title="Notifications"
+          badge={!isNotified?getFollows().length:null} 
+          /* badgeColor="red" */
+          onPress={() => {
+            setIsNotified(!isNotified?!isNotified:isNotified);
```

Comment:
You can abbreviate this to `setIsNotified(!isNotifed)`

Reply:
Ok. Will make the change.

Reply:
`setIsNotified(!notified)` wont work as expected if the user switches tabs (i.e. taps on Notifications) every other round since the value will cycle through `true` and `false` with each tap.

`setIsNotified(!notified?!notified:notified)` will work however. Logically however `setIsNotified(!notified)` is correct but because it is simple state implementation for a visual experience, this quirk is introduced.

Reply:
Good point. Should just be `setIsNotified(true)`

Reply:
That should have been obvious. Will make that change.